File: src/plugins/meson-templates/resources/src/window-gtk4.py
{{include "license.py"}}

{{if is_adwaita}}
from gi.repository import Adw
{{end}}
from gi.repository import Gtk

@Gtk.Template(resource_path='{{appid_path}}/{{ui_file}}')
class {{PreFix}}Window({{if is_adwaita}}Adw{{else}}Gtk{{end}}.ApplicationWindow):
    __gtype_name__ = '{{PreFix}}Window'

    label = Gtk.Template.Child()

    def __init__(self, **kwargs):
        super().__init__(**kwargs)

File: src/plugins/meson-templates/resources/src/window-gtk4.js
{{include "license.js"}}

import GObject from 'gi://GObject';
import Gtk from 'gi://Gtk';
{{if is_adwaita}}
import Adw from 'gi://Adw';
{{end}}

export const {{PreFix}}Window = GObject.registerClass({
    GTypeName: '{{PreFix}}Window',
    Template: 'resource://{{appid_path}}/{{ui_file}}',
    InternalChildren: ['label'],
}, class {{PreFix}}Window extends {{if is_adwaita}}Adw{{else}}Gtk{{end}}.ApplicationWindow {
    constructor(application) {
        super({ application });
    }
});

